src/rsz/src/Resizer.cc

```diff
@@ -2488,55 +2493,88 @@ Resizer::repairHoldPass(VertexSet &hold_failures,
+            if (fanout_port)
+              load_cap += fanout_port->capacitance();
           }
         }
       }
       if (!load_pins.empty()) {
-        Delay buffer_delay = bufferHoldDelay(buffer_cell);
-        int buffer_count = std::ceil(insert_delay / buffer_delay);
-        debugPrint(logger_, RSZ, "repair_hold", 2,
-                   " {} hold={} inserted {} for {}/{} loads",
-                   vertex->name(sdc_network_),
-                   delayAsString(hold_slack, this, 3),
-                   buffer_count,
-                   load_pins.size(),
-                   fanout(vertex));
-        makeHoldDelay(vertex, buffer_count, load_pins,
-                      loads_have_out_port, buffer_cell);
-        repair_count += buffer_count;
-        if (inserted_buffer_count_ > max_buffer_count
-            || overMaxArea())
-          return repair_count;
+        // multi-corner support MIA
+        const DcalcAnalysisPt *dcalc_ap = sta_->cmdCorner()->findDcalcAnalysisPt(max_);
+        Delay buffer_delay1 = bufferDelay(buffer_cell, drvr_rf, load_cap, dcalc_ap);
+        // Need enough slack to at least insert the last buffer with loads.
+        if (allow_setup_violations
+            || (buffer_delay1 < drvr_slacks[RiseFall::riseIndex()][max_index]
+                && buffer_delay1 < drvr_slacks[RiseFall::fallIndex()][max_index])) {
+          Delay buffer_delay = bufferHoldDelay(buffer_cell);
+          Delay max_insert = min(drvr_slacks[RiseFall::riseIndex()][max_index],
+                                 drvr_slacks[RiseFall::fallIndex()][max_index]);
+          int max_insert_count = (max_insert - buffer_delay1) / buffer_delay + 1;
+          int hold_buffer_count = (-drvr_hold_slack > buffer_delay1)
+            ? std::ceil((-drvr_hold_slack-buffer_delay1)/buffer_delay)+1
+            : 1;
+          int buffer_count = allow_setup_violations
+            ? hold_buffer_count
+            : min(max_insert_count, hold_buffer_count);
+          debugPrint(logger_, RSZ, "repair_hold", 2,
+                     " {} hold={} inserted {} for {}/{} loads",
+                     vertex->name(sdc_network_),
+                     delayAsString(drvr_hold_slack, this, 3),
+                     buffer_count,
+                     load_pins.size(),
+                     fanout(vertex));
+          makeHoldDelay(vertex, buffer_count, load_pins,
+                        loads_have_out_port, buffer_cell);
+          repair_count += buffer_count;
+#if 0
+          // Check that no setup violations are introduced.
+          ensureWireParasitics();
+          Slack drvr_setup_slack1 = sta_->vertexSlack(vertex, max_);
+          if (drvr_setup_slack1 < 0
+              && drvr_setup_slack1 < drvr_setup_slack)
+            printf("%s %s -> %s\n", vertex->name(network_),
+                   delayAsString(drvr_setup_slack, sta_, 3),
+                   delayAsString(drvr_setup_slack1, sta_, 3));
+#endif
```

Comment:
Is this code blocked by some TODO, or is it just dead code? If it's the latter do you think it would be better to remove it?

Reply:
It is a check that I use for debugging